zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  # Build a second bootloader image
+
+  set(MCUBOOT_BASE ${CMAKE_CURRENT_LIST_DIR}/..)
+
+  zephyr_add_executable(mcuboot require_build)
+
+  if (${require_build})
+    add_subdirectory(${MCUBOOT_BASE}/boot/zephyr ${CMAKE_CURRENT_BINARY_DIR}/mcuboot)
+
+    # TODO: Assert that the bootloader and image use the same key.
+
+    set(SIGNED_IMAGE signed.hex)
+
+    set_property(GLOBAL APPEND PROPERTY
+      extra_post_build_commands
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${MCUBOOT_BASE}/scripts/imgtool.py
+      sign
+      --key ${MCUBOOT_BASE}/${CONFIG_BOOT_SIGNATURE_KEY_FILE}
+      --header-size ${CONFIG_TEXT_SECTION_OFFSET}
+      --align       ${DT_FLASH_WRITE_BLOCK_SIZE}
+      --version 0.1       # TODO: Configurable?
```

Comment:
> TODO: Configurable?

I think so. One could set it at CMake time; maybe even let the environment override that. That'd be pretty convenient on macOS and Linux at least.

Reply:
Could you elaborate the benefits of not having this as a Kconfig option?

Reply:
I'm thinking of things like developer workflows where you want to update this on a build-by-build basis every few minutes, without having to go through the overhead of running menuconfig. An environment variable makes that pretty easy -- not sure if it's possible to have a kconfig option fall back to an env var dynamically, but I believe not. OTOH from a reproducible builds perspective I take your point. Tradeoffs here, your call.